frigate/object_detection.py

```diff
@@ -54,19 +52,12 @@ def __init__(
         else:
             self.input_transform = None
 
-        if det_type == DetectorTypeEnum.edgetpu:
-            self.detect_api = EdgeTpuTfl(
-                det_device=det_device, model_config=model_config
-            )
-        elif det_type == DetectorTypeEnum.openvino:
-            self.detect_api = OvDetector(
-                det_device=det_device, model_config=model_config
-            )
-        else:
+        if det_type == DetectorTypeEnum.cpu:
             logger.warning(
                 "CPU detectors are not recommended and should only be used for testing or for trial purposes."
             )
-            self.detect_api = CpuTfl(model_config=model_config, num_threads=num_threads)
+
+        self.detect_api = DetectionApi.create(det_type, det_device=det_device, model_config=model_config, num_threads=num_threads)
```

Comment:
I'm wondering if there's a better way to do this, if there are detectors in the future that require other args this list could get longer even if most of the args are not relevant to each detector.

Reply:
I was thinking about this as well. I think the best way to do this would be to pass in the `detectors` config as a model, and let the detector itself suss out the args from that. I wasn't sure if I should go that deep into it before knowing whether even this much is beneficial or not though.

Reply:
Another thing that was discussed was reorganizing the model configuration to be under the detector, so we can have different models if we have more than one detector type.  That might make it easier to bundle up some of the detector config into a single arg to pass into the create function.

Reply:
Should I try and take at stab at reorganizing the model along with this pull request? Or is it better for a separate pull request?

Reply:
Up to you